runtime/src/main/java/io/micronaut/discovery/cloud/ComputeInstanceMetadataResolverUtils.java

```diff
@@ -64,35 +67,26 @@ public static JsonNode readMetadataUrl(URL url, int connectionTimeoutMs, int rea
             uc.setReadTimeout(readTimeoutMs);
             uc.setRequestMethod(HttpMethod.GET.name());
             uc.setDoOutput(true);
-            try (InputStream in = uc.getInputStream()) {
-                return objectMapper.readTree(in);
+            try (InputStream in = uc.getInputStream();
+                 JsonParser parser = jsonFactory.createParser(in)) {
+                return treeCodec.readTree(parser);
             }
         }
     }
 
-    /**
-     * Resolve a value as a string from the metadata json.
-     * @param json The json
-     * @param key The key
-     * @return An optional value
-     */
-    public static Optional<String> stringValue(JsonNode json, String key) {
-        return Optional.ofNullable(json.findValue(key)).map(JsonNode::asText);
-    }
-
     /**
      * Populates the instance instance metadata's {@link AbstractComputeInstanceMetadata#setMetadata(Map)} property.
+     *
      * @param instanceMetadata The instance metadata
-     * @param metadata A map of metadata
+     * @param metadata         A json object of metadata
      */
-    public static void populateMetadata(AbstractComputeInstanceMetadata instanceMetadata, Map<?, ?> metadata) {
+    public static void populateMetadata(AbstractComputeInstanceMetadata instanceMetadata, JsonNode metadata) {
```

Comment:
These methods seem to be used in modules and changing them will break compatibility. See:

https://github.com/micronaut-projects/micronaut-oracle-cloud/blob/ee5fe9110797e473c044ae20c36ee8987e2929bf/oraclecloud-common/src/main/java/io/micronaut/discovery/cloud/oraclecloud/OracleCloudMetadataResolver.java#L42-L43

Reply:
ok i will readd the old methods for now and mark the new ones as experimental. once the api has settled, we can migrate depending modules to the new methods.